File: Algorithm/baekjoon/java/BJ11047.java
package baekjoon.java;

import java.util.*;


public class BJ11047 {
    static int minimumCoin(int n, int k, int[] values){
        int money = k;
        int coins = 0;

        for(int i = n - 1; i >= 0; --i){
            while(money >= values[i]){
                money -= values[i];
                coins += 1;
            }
            if(money <= 0) break;
        }

        return coins;
    }

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);

        int n = scanner.nextInt();
        int k = scanner.nextInt();
        int[] values = new int[n];

        for(int i = 0; i < n; ++i){
            values[i] = scanner.nextInt();
        }

        int output = minimumCoin(n, k, values);

        System.out.println(output);
    }
}

File: Algorithm/baekjoon/swift/bj11047.swift

func minimumCoin(money: Int, coins: [Int]) -> Int{
    var output : Int = 0
    var input = money
    var length = coins.count;
    for i in length...0{
        output += money/coins[i]
        input %= coins[i]
    }
    return output
}
var n = Int(readLine()!)!;
var money = Int(readLine()!)!
var coins : [Int] = []

for i in 0..<n {
    coins[i] = Int(readLine()!)!
}

var output = minimumCoin(money: money, coins: coins)
print(output)